onnxruntime/core/providers/qnn/qnn_execution_provider.h

```diff
@@ -23,6 +23,59 @@ namespace onnxruntime {
 
 void RunOnUnload(std::function<void()> function);
 
+class SharedContext {
+ public:
+  static SharedContext& GetInstance() {
+    static SharedContext instance_;
+    return instance_;
+  }
+
+  bool HasSharedQnnModels() {
+    const std::lock_guard<OrtMutex> lock(mtx_);
+    return !shared_qnn_models_.empty();
+  }
+
+  std::shared_ptr<qnn::QnnModel> GetSharedQnnModel(const std::string& model_name) {
+    const std::lock_guard<OrtMutex> lock(mtx_);
+    auto it = find_if(shared_qnn_models_.begin(), shared_qnn_models_.end(),
+                      [&model_name](const std::shared_ptr<qnn::QnnModel>& qnn_model) { return qnn_model->Name() == model_name; });
+    if (it == shared_qnn_models_.end()) {
+      return nullptr;
+    }
+    return *it;
+  }
+
+  bool SetSharedQnnModel(std::vector<std::shared_ptr<qnn::QnnModel>>& shared_qnn_models,
```

Comment:
Could this parameter be made a const reference? It doesn't seem to be modified by this function.

Reply:
Still thinking that maybe should change shared_ptr to unique_ptr to make the ownership clear.